OutOfMemoryError in ImageHelper

Found another place where OutOfMemoryError needs to be caught:

```
03-26 20:33:01.310    5058-5094/org.wordpress.android E/AndroidRuntime﹕ FATAL EXCEPTION: AsyncTask #3
    java.lang.RuntimeException: An error occured while executing doInBackground()
            at android.os.AsyncTask$3.done(AsyncTask.java:278)
            at java.util.concurrent.FutureTask$Sync.innerSetException(FutureTask.java:273)
            at java.util.concurrent.FutureTask.setException(FutureTask.java:124)
            at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:307)
            at java.util.concurrent.FutureTask.run(FutureTask.java:137)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:208)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
            at java.lang.Thread.run(Thread.java:856)
     Caused by: java.lang.OutOfMemoryError
            at android.graphics.BitmapFactory.nativeDecodeStream(Native Method)
            at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:493)
            at android.graphics.BitmapFactory.decodeFile(BitmapFactory.java:299)
            at org.wordpress.android.util.ImageHelper.createThumbnailFromUri(ImageHelper.java:422)
            at org.wordpress.android.util.PostUploadService$UploadPostTask.uploadMediaFile(PostUploadService.java:627)
            at org.wordpress.android.util.PostUploadService$UploadPostTask.doInBackground(PostUploadService.java:256)
            at org.wordpress.android.util.PostUploadService$UploadPostTask.doInBackground(PostUploadService.java:130)
            at android.os.AsyncTask$2.call(AsyncTask.java:264)
            at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
            at java.util.concurrent.FutureTask.run(FutureTask.java:137)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:208)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
            at java.lang.Thread.run(Thread.java:856)
```

--- a/src/org/wordpress/android/util/ImageHelper.java
+++ b/src/org/wordpress/android/util/ImageHelper.java
@@ -405,7 +405,12 @@ public class ImageHelper {
         BitmapFactory.Options optBounds = new BitmapFactory.Options();
         optBounds.inJustDecodeBounds = true;
 
-        BitmapFactory.decodeFile(filePath, optBounds);
+        try {
+            BitmapFactory.decodeFile(filePath, optBounds);
+        } catch (OutOfMemoryError e) {
+            WPMobileStatsUtil.trackEventForSelfHostedAndWPCom(WPMobileStatsUtil.StatsEventMediaOutOfMemory);
+            return null;
+        }
 
         // determine correct scale value (should be power of 2)
         // http://stackoverflow.com/questions/477572/android-strange-out-of-memory-issue/3549021#3549021
@@ -419,7 +424,14 @@ public class ImageHelper {
         optActual.inSampleSize = scale;
 
         // Get the roughly resized bitmap
-        Bitmap bmpResized = BitmapFactory.decodeFile(filePath, optActual);
+        Bitmap bmpResized;
+        try {
+            bmpResized = BitmapFactory.decodeFile(filePath, optActual);
+        } catch (OutOfMemoryError e) {
+            WPMobileStatsUtil.trackEventForSelfHostedAndWPCom(WPMobileStatsUtil.StatsEventMediaOutOfMemory);
+            return null;
+        }
+
         if (bmpResized == null)
             return null;
 

